include/exadg/time_integration/time_int_bdf_base.h

```diff
@@ -101,6 +104,49 @@ class TimeIntBDFBase : public TimeIntBase
   double
   get_previous_time(int const i /* t_{n-i} */) const;
 
+  /*
+   * Get current order of time integrator
+   */
+  unsigned int
+  get_current_order() const;
+
+  /*
+   * Fill a vector at previous times given a lambda.
+   */
+  template<typename Iterator, typename Lambda>
+  void
+  fill_at_previous_times(Iterator const & begin, Iterator const & end, Lambda const & f) const
+  {
+    /*
+     *
+     *   time t
+     *  -------->   t_{n-2}   t_{n-1}   t_{n}     t_{n+1}
+     *  _______________|_________|________|___________|___________\
+     *                 |         |        |           |           /
+     *                end       ...     begin
+     */
+
+    AssertThrow(std::distance(begin, end) <= get_current_order(),
+                dealii::ExcMessage("Iterator range exceeds available time steps."));
+
+    unsigned int i = 0;
+    for(auto it = begin; it != end; ++it, ++i)
+      *it = f(i);
+  }
```

Comment:
This function makes the code cryptic in my opinion. Note that "previous times" in the documentation above does not occur in the implementation of the function.

Reply:
Removed in [1f39cef](https://github.com/exadg/exadg/pull/458/commits/1f39cef8d688232b73b239620a45f409a169e862)